code/modules/mob/living/carbon/xenomorph/castes/Queen.dm

```diff
@@ -855,3 +838,19 @@
 
 /mob/living/carbon/Xenomorph/Queen/gib(var/cause = "gibbing")
 	death(cause, 1)
+
+/datum/behavior_delegate/queen
+	name = "Queen Behavior Delegate"
+
+/datum/behavior_delegate/queen/on_update_icons()
+	if(bound_xeno.stat == DEAD)
+		return
+
+	var/mob/living/carbon/Xenomorph/Queen/Q = bound_xeno
+	if(Q.ovipositor)
+		Q.icon = Q.queen_ovipositor_icon
+		Q.icon_state = "[Q.mutation_type] Queen Ovipositor"
+		return TRUE
+
+	// Switch icon back and then let normal icon behavior happen
+	Q.icon = Q.queen_standing_icon
```

Comment:
Use a more descriptive variable name here, it shouldn't be Q it should be 'queen'